### PR TITLE
n3 Parser type fix

### PR DESCRIPTION
**Pull request by stuarthendren**
The [n3](https://github.com/rdfjs/N3.js) parser now accepts two callbacks, and a separate inline prefix callback.

https://github.com/rdfjs/N3.js/blob/42ab3a29e9ee82cf4d8fdf8994c1f7e74f9b1035/src/N3Parser.js#L982

My understanding is:
If giving a parse callback the function is async and the prefix callback is optional.

If the parse callback is not supplied or nullish, the parse is sync and returns the quads. 
Supplying null and a prefix callback also works synchronously and calls the prefix callback
while processing. (Not sure if this is intended, but it is useful).

I have updated the version as I assume this has changed at some point. I have not checked for other API changes.

Update: after being prompted, I have also added the other missing types. This maybe not be absolutely correct as I have not checked all the other existing types but is closer to the current version. 

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by stuarthendren** [comment]
The tests have helpfully pointed me to other changes in the API. I'll try to add typings for these shortly.

**Comment by stuarthendren** [comment]
Added missing types for the Lexer and the term/id until function.

File: types/n3/index.d.ts [hunk 1]
```diff
@@ -130,6 +130,24 @@ export interface BlankTriple<Q extends RDF.BaseQuad = RDF.Quad> {
...
+    lineMode?: boolean;
+    n3?: boolean;
+    comments?: boolean;
+}
+
+export type TokenCallback = (error: Error, token: Token) => void;
+
+export class Lexer {
+    constructor(options?: LexerOptions);
+    tokenize(input: string | EventEmitter): Token[];
```
Comment (joachimvh) [inline types/n3/index.d.ts:148]:
```suggestion
    tokenize(input: string): Token[];
```
A callback seems to be required when the input is an EventEmitter: https://github.com/rdfjs/N3.js/blob/89f0daf748bf2d6ca194b37e58733798346a88e0/src/N3Lexer.js#L455-L485

File: types/n3/index.d.ts [hunk 2]
```diff
@@ -130,6 +130,24 @@ export interface BlankTriple<Q extends RDF.BaseQuad = RDF.Quad> {
     object: Q['object'];
 }
 
+export interface Token {
+    type: string;
+    value?: string;
+    line: number;
+}
```
Comment (joachimvh) [inline types/n3/index.d.ts:137]:
```suggestion
    prefix?: string;
}
```

Tokens can also have a prefix field: https://github.com/rdfjs/N3.js/blob/89f0daf748bf2d6ca194b37e58733798346a88e0/src/N3Lexer.js#L348

**Review by joachimvh — CHANGES_REQUESTED** [review]
Sorry for the delay. I had a look into the Lexer code to check the new typings and it seems there are some minor things where the typings differ from what happens. Besides that most of it seems fine.

**Comment by stuarthendren** [comment]
> Sorry for the delay. I had a look into the Lexer code to check the new typings and it seems there are some minor things where the typings differ from what happens. Besides that most of it seems fine.

No worries. I've added those suggestions. Thanks for reviewing.

**Review by joachimvh — APPROVED** [review]

**Comment by stuarthendren** [comment]
Ready to merge

**Pull request merged**